profiles.md

```diff
@@ -0,0 +1,234 @@
+# Researcher profiles Endpoints
+[Back to the list of all defined endpoints](endpoints.md)
+
+This repository allows to manage the profile item linked to a particular eperson. The specific type of item 
+with which the profile is modeled depends on the dspace configuration (by default it is the Person type). If the particular dspace instance doesn't handle the "profile" 
+type then the whole repository is disabled.
+
+## Get all Profile
+**GET /api/eperson/profiles/**
+
+This operation is not currently supported.
+
+## Single Profile
+**GET /api/eperson/profiles/<:eperson-uuid>**
+
+Provide information about a specific profile. Only the Administrators or the owner eperson can access this endpoint.
+
+The JSON response document is as follow
+
+```json
+{
+  "id": "eb645ef8-1373-41eb-bf67-6afcea7e2069",
+  "visible": true,
+  "type": "profile"
+}
+```
+
+Attributes:
+* id: the UUID of the EPerson who owns the profile. This also corresponds to the ID of the profile.
+* visible: indicates whether the profile is public (visible: true) or private (otherwise). If private only 
```

Comment:
Can the visible property be dropped?

Permissions are handled by item resource policies, and having a boolean can make it behave strangely and hard to keep in sync (especially if the resource policies make it neither public or private, but a permission in between (e.g. staff group read)

Reply:
I agree with @benbosman 's feedback here.  I did also want to add that we *do* have a concept of a "discoverable" or "non-discoverable" Item as well which is also similar, but that's about whether the Item is *indexed* or not.

So, I agree this "visible" attribute is a bit confusing after looking at it again.   It should likely either be replaced by "discoverable / non-discoverable" or resource policies...we shouldn't be building a different way to access restrict a profile.

Reply:
The visible attribute indicates only whether the profile (ie the item that models it) is visible or not to anonymous users. Internally, the visibility of the item is already managed with the resource policies, and in fact this value is true if the Anonymous group has the permissions to read on the anonymous user and false if it does not. A patch in which the visibility is set to true therefore means to add the read policy to the anonymous group while setting the value to false means to remove it.

The posibility to change the profile visibility using this attribute is not strictly necessary for orcid, so if you are not convinced we can also not bring it to dspace. As you prefer.

Reply:
@LucaGiamminonni : I'd like to hear @benbosman's thoughts on this as well, but one approach here could be to make it clearer what the "visible" property *does*.  If it's just a quick way to add/remove an Anonymous READ resource policy, then we should say that in the description in this REST Contract.  Currently, it implies it does more than that, as it says setting it to "false" makes it Administrator Only (But that's not entirely true, as removing the Anonymous READ policy just means it's no longer available anonymously, but other policies might specify a different permission level than Admin-only)

So, in my opinion...

* Either, we need to fix this REST Contract description to say something like 
`visible:  This property provides a quick way to add/remove the Anonymous READ policy. When set to "true", then the Anonymous READ Policy is added (and the profile becomes accessible to anonymous users). When set to "false", then the Anonymous READ policy is removed (and the profile is no longer accessible anonymously).`
* Or, we should just remove it entirely and require that you add/remove policies on the Item using the `/api/authz/resourcepolicies` endpoint.

Reply:
I can change the description of the visible attribute but I prefer to keep visible instead of public, even cause "public" is a reserved word.

Reply:
@LucaGiamminonni it's good you removed the wording which claims to know who has permissions if not `visible`.
If we'd have to keep this, I would recommend to further clarify this since this is a very atypical property. This can be done by stating the permissions are unknown if `visible=false`. It may be accessible to many users, or admins only, or also the EPerson itself (if they're granted READ permissions), …. The only thing that's known is whether there's an Anonymous READ policy without start/end date